Add a mobile-friendly "About" button, as well as
style fixes for smaller width screens on built-in pages

--- a/browser/base/content/appmenu-viewcache.inc.xhtml
+++ b/browser/base/content/appmenu-viewcache.inc.xhtml
@@ -154,6 +154,10 @@
                      class="subviewbutton subviewbutton-nav"
                      data-l10n-id="appmenuitem-help"
                      closemenu="none"
+                     />
+      <toolbarbutton id="appMenu-about-button"
+                     class="subviewbutton"
+                     data-l10n-id="menu-about"
                      />
 #ifndef XP_MACOSX
       <toolbarseparator/>
--- a/browser/components/customizableui/content/panelUI.js
+++ b/browser/components/customizableui/content/panelUI.js
@@ -385,6 +385,9 @@
       case "appMenu-help-button2":
         this.showSubView("PanelUI-helpView", target);
         break;
+      case "appMenu-about-button":
+        openTrustedLinkIn("chrome://browser/content/aboutDialog.xhtml", "tab");
+        break;
     }
   },
 
--- a/browser/base/content/aboutDialog.css
+++ b/browser/base/content/aboutDialog.css
@@ -169,3 +169,24 @@
     content: url("chrome://global/skin/icons/loading.svg");
   }
 }
+
+@media (max-width: 600px) {
+  #clientBox {
+    flex-direction: column;
+  }
+  #leftBox {
+    margin-inline-start: 0;
+  }
+  vbox#rightBox {
+    width: auto;
+  }
+  label.bottom-link {
+    margin: 0;
+    flex: 1;
+    align-content: center;
+  }
+}
+
+#aboutDialogContainer {
+  overflow: scroll;
+}
